.github/workflows/reusable-tests.yml

```
@@ -121,10 +135,30 @@ jobs:
         inputs.dist-type == 'source' &&
         runner.os == 'Linux'
       run: sudo apt update && sudo apt install build-essential libssl-dev
-    - name: Install libssh and openssl headers on macOS
+    - name: (macOS) Download universal2 OpenSSL+libssh
       if: >-
-        runner.os == 'macOS'
-      run: brew install libssh
+        runner.os == 'macOS' && inputs.macos-universal-artifact-name != ''
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ inputs.macos-universal-artifact-name }}
+        path: .deps/
+    - name: (macOS) Unpack deps and export build env
+      if: >-
+        runner.os == 'macOS' && inputs.macos-universal-artifact-name != ''
```

This conditional should probably be
```suggestion
        runner.os == 'macOS' && inputs.dist-type == 'source'
```
instead.